### PR TITLE
DOC: Use `spin` to generate changelog

### PR DESCRIPTION
**Pull request by ganesh-k13**
### Changes
- Use `spin` to generate changelog

### Notes
Follow up on #24986, I missed modifying the docs

### CODE REVIEW

File: doc/RELEASE_WALKTHROUGH.rst [hunk 1]
```diff
@@ -79,7 +79,7 @@ Generate the changelog
 
 The changelog is generated using the changelog tool::
 
-    $ python tools/changelog.py $GITHUB v1.20.0..maintenance/1.21.x > doc/changelog/1.21.0-changelog.rst
+    $ spin changelog -t $GITHUB --revision-range v1.20.0..maintenance/1.21.x > doc/changelog/1.21.0-changelog.rst
```
Comment (charris) [inline doc/RELEASE_WALKTHROUGH.rst:82]: I haven't tried this yet, but are `-t` and `--revision-range` optional of required?

Comment (ganesh-k13) [reply doc/RELEASE_WALKTHROUGH.rst:82]: It's required. The script it calls downstream need the args: https://github.com/numpy/numpy/blob/44efd35ab74b8aac6b580a927b0d540344551610/tools/changelog.py#L109

We can make the token a default value env variable, but the revision must be required. We can infer it however to default to n-1 and nth release, where n is latest.

Comment (charris) [reply doc/RELEASE_WALKTHROUGH.rst:82]: The script specifies them as positional arguments. Do the spin commands allow that?

Comment (ganesh-k13) [reply doc/RELEASE_WALKTHROUGH.rst]: Yep, indeed. We need to change `option` to `argument`. I have made the changes.

**Comment by ngoldbaum** [comment]
@charris ok with merging this one now? I don't want to mess with your release workflow.

**Comment by charris** [comment]
Thanks Ganesh.

> ok with merging this one now

Yes, but I am just now adding an optional argument to changelog :)  I need to remember to update this.

**Pull request merged**